app/src/main/java/org/android/go/sopt/presentation/auth/SignUpViewModel.kt

```diff
@@ -0,0 +1,77 @@
+package org.android.go.sopt.presentation.auth
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.Transformations
+import androidx.lifecycle.ViewModel
+import org.android.go.sopt.data.remote.SignUpRequestDTO
+import org.android.go.sopt.data.remote.SignUpResponseDTO
+import org.android.go.sopt.module.AuthServicePool
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+import java.util.regex.Pattern
+
+class SignUpViewModel : ViewModel() {
+    private val _signUpResult: MutableLiveData<SignUpResponseDTO> = MutableLiveData()
+    val signUpResult: LiveData<SignUpResponseDTO> = _signUpResult
+
+    private val _errorResult: MutableLiveData<String> = MutableLiveData()
+    val errorResult: LiveData<String> = _errorResult
+
+    val idText: MutableLiveData<String> = MutableLiveData("")
+    val pwText: MutableLiveData<String> = MutableLiveData("")
+    val nameText: MutableLiveData<String> = MutableLiveData("")
+    val skillText: MutableLiveData<String> = MutableLiveData("")
+
+    val isIdValid: LiveData<Boolean> = Transformations.map(idText) { checkIdValid(it) }
+    val isPwValid: LiveData<Boolean> = Transformations.map(pwText) { checkPwValid(it) }
```

Comment:
```suggestion
    val isIdValid: LiveData<Boolean> = Transformations.map(idText) { id -> checkIdValid(id) }
    val isPwValid: LiveData<Boolean> = Transformations.map(pwText) { pw -> checkPwValid(pw) }
```
스코프 함수에서 `it` 대신 더 명시적인 네이밍을 작성해주면 코드 가독성이 좋아질 것 같습니당